src/client/Microsoft.Identity.Client/Platforms/Features/RuntimeBroker/RuntimeBroker.cs

```diff
@@ -312,6 +312,21 @@ private IntPtr GetParentWindow(CoreUIParent uiParent)
                         var errorMessage = "Could not acquire token silently.";
                         msalTokenResponse = WamAdapters.HandleResponse(result, authenticationRequestParameters, _logger, errorMessage);
                     }
+
+                    if (acquireTokenSilentParameters.ForceRefresh && !string.IsNullOrEmpty(msalTokenResponse.AccessToken))
+                    {
+                        authParams.AccessTokenToRenew = msalTokenResponse.AccessToken;
```

Comment:
Out of curiosity, under what circumstance will a mandatory refresh become necessary?

All native MSALs have `.WithForceRefresh(...)` but such an api risks being abuse to always bypass token cache. OneAuth/MsalRuntime's `AccessTokenToRenew` is a better design which does not have the same risk. If we are making a change here anyway, I would suggest to consider adding a new `.WithAccessTokenToRenew(...)` design to native MSALs.

Reply:
The abuse of this force refresh was discussed today and concluded eSTS will throttle in care of abuse. That said, this option exists today. Are you suggesting to remove it?

Reply:
Yes, I totally agree with abuse part and currently the plan is to unify the programing model between MSAL.CPP and native MSALs in future releases but the decision for this particular fix was to use the existing paradigm.

For the investigation I am working through a fresh AT is needed when user went and removed the account from broker and added it again to the broker this way re-provisioning the device with a new device id. The issue is the cached AT has the wrong device-id claim.

Reply:
> For the investigation I am working through a fresh AT is needed when user went and removed the account from broker and added it again to the broker this way re-provisioning the device with a new device id. The issue is the cached AT has the wrong device-id claim.

Legit use case. And it is indeed a case that the existing native MSAL's WithForceRefresh() is difficult to handle. It relies on the app developers to call WithForceRefreesh() only when necessary.


> this option exists already in public API surface. Are you suggesting to remove it?

No. Never remove existing API unless we are willing to bump major version number.


> decision for this particular fix was to use the existing paradigm.

If this time is not a good timing and good reason to naturally add the new paradigm, I don't know when it will be. :-)